internal/credentials/kubernetes/ecr/pod_identity.go

```diff
@@ -82,12 +83,17 @@ func (p *podIdentityCredentialHelper) getCredentials(
 	repoURL string,
 	_ *corev1.Secret,
 ) (*credentials.Credentials, error) {
-	if credType != credentials.TypeImage ||
+	if credType == credentials.TypeGit ||
```

Comment:
```suggestion
	if (credType != credentials.TypeImage && credType != credentials.TypeHelm) ||
```

Similar to https://github.com/akuity/kargo/pull/2280/files#r1674339382